new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java

```diff
@@ -214,156 +224,281 @@ public void testCreateMiiDomain() {
+            new OperatorParams()
+                    .helmParams(opHelmParams)
+                    .image(OPERATOR_IMAGE)
+                    .domainNamespaces(Arrays.asList(domainNamespace,domainNamespace1))
+                    .serviceAccount(serviceAccountName);
+
+    // install Operator
+    logger.info("Upgrading Operator in namespace {0}", opNamespace);
+    assertTrue(upgradeOperator(opParams),
+            String.format("Operator upgrade failed in namespace %s", opNamespace));
+    logger.info("Operator upgraded in namespace {0}", opNamespace);
+
+    // Create the repo secret to pull the image
+    assertDoesNotThrow(() -> createRepoSecret(domainNamespace1),
+            String.format("createSecret failed for %s", repoSecretName));
+
+    // create secret for admin credentials
+    logger.info("Create secret for admin credentials");
+    String adminSecretName = domainUID1 + "-weblogic-credentials";
+    assertDoesNotThrow(() -> createDomainSecret(adminSecretName,"weblogic",
+            "welcome2", domainNamespace1),
+            String.format("createSecret failed for %s", adminSecretName));
+
+    // create encryption secret
+    logger.info("Create encryption secret");
+    String encryptionSecretName = "encryptionsecretdomain2";
+    assertDoesNotThrow(() -> createDomainSecret(encryptionSecretName, "weblogicencdomain2",
+            "weblogicencdomain2", domainNamespace1),
+             String.format("createSecret failed for %s", encryptionSecretName));
+
+    // create the domain CR
+    int adminNodePort = 30712;
+    createDomainResource(domainUID1, domainNamespace1, adminSecretName, repoSecretName,
+            encryptionSecretName, adminNodePort, replicaCount);
+
+    // wait for the domain to exist
+    logger.info("Check for domain custom resouce in namespace {0}", domainNamespace1);
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Waiting for domain {0} to be created in namespace {1} "
+                    + "(elapsed time {2}ms, remaining time {3}ms)",
+                domainUID1,
+                domainNamespace1,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(domainExists(domainUID1, DOMAIN_VERSION, domainNamespace1));
+
+
+    // check admin server pod exist
+    logger.info("Check for admin server pod {0} existence in namespace {1}",
+            adminServerPodName, domainNamespace1);
+    checkPodCreated(adminServerPodName, domainUID1, domainNamespace1);
+
+    // check managed server pods exists
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Check for managed server pod {0} existence in namespace {1}",
+              managedServerPrefix + i, domainNamespace1);
+      checkPodCreated(managedServerPrefix + i, domainUID1, domainNamespace1);
+    }
+
+    // check admin server pod is running
+    logger.info("Wait for admin server pod {0} to be ready in namespace {1}",
```

Comment:
ready or running?

Reply:
This is also existing code. But I looked into it and the code only checks if the pod is Running - not if it is ready to service request. Lenny is making changes to Kubernetes.java and he is going to add a method there to check if the pod is ready.

Reply:
Actually, Xian has already made the change in her PR - to check Pod is ready instead of just Running.